### PR TITLE
Update docker-device-tool

### PR DESCRIPTION
**Pull request by alexlarsson**
This makes docker-device-tool build again and adds a few new commands:
- status: Get pool status
- list: List all devices
- device: Get device status
- resize: resize the loopback devices and pool

### CODE REVIEW

**Comment by crosbymichael** [comment]
LGTM

ping @creack

File: graphdriver/devmapper/deviceset.go [hunk 1]
```diff
@@ -357,13 +367,90 @@ func minor(device uint64) uint64 {
...
+	if metadatafile == nil {
+		return err
+	}
+	defer metadatafile.Close()
+
+	metadataloopback := dmFindLoopDeviceFor(metadatafile)
+	if metadataloopback == nil {
+		return fmt.Errorf("Unable to find loopback mount for: %s", metadatafilename)
+	}
+	defer metadataloopback.Close ()
```
Comment (creack) [inline graphdriver/devmapper/deviceset.go]: Can you run gofmt on the commit?

File: graphdriver/devmapper/deviceset.go [hunk 2]
```diff
@@ -757,6 +846,76 @@ func (devices *DeviceSet) setInitialized(hash string) error {
 	return nil
 }
 
+func (devices *DeviceSet) List() []string {
+	devices.Lock()
+	defer devices.Unlock()
+
+	ids := make([]string, len(devices.Devices))
+	i := 0
+	for k, _ := range devices.Devices {
```
Comment (creack) [inline graphdriver/devmapper/deviceset.go]: gofmt might skip this. Please apply `gofmt -w -s`

File: graphdriver/devmapper/deviceset.go [hunk 3]
```diff
@@ -757,6 +846,76 @@ func (devices *DeviceSet) setInitialized(hash string) error {
...
+		ids[i] = k
+		i++
+	}
+	return ids
+}
+
+func (devices *DeviceSet) deviceStatus(devName string) (sizeInSectors, mappedSectors, highestMappedSector uint64, err error) {
+	var params string
+	fmt.Println("devname", devName)
```
Comment (creack) [inline graphdriver/devmapper/deviceset.go]: Can you avoid to print on Stdout directly? Prefer utils.Debugf or store a custom Stdout.

File: graphdriver/devmapper/deviceset.go [hunk 4]
```diff
@@ -757,6 +846,76 @@ func (devices *DeviceSet) setInitialized(hash string) error {
...
+	_, sizeInSectors, _, params, err = getStatus(devName)
+	if err != nil {
+		return
+	}
+	fmt.Println("params", params)
```
Comment (creack) [inline graphdriver/devmapper/deviceset.go]: Same here

File: graphdriver/devmapper/deviceset.go [hunk 5]
```diff
@@ -757,6 +846,76 @@ func (devices *DeviceSet) setInitialized(hash string) error {
...
+	status.SizeInSectors = sizeInSectors
+	status.MappedSectors = mappedSectors
+	status.HighestMappedSector = highestMappedSector
+
+	return status, nil
+}
+
+func (devices *DeviceSet) poolStatus() (totalSizeInSectors, transactionId, dataUsed, dataTotal, metadataUsed, metadataTotal uint64, err error) {
+	var params string
+	_, totalSizeInSectors, _, params, err = getStatus(devices.getPoolName())
```
Comment (creack) [inline graphdriver/devmapper/deviceset.go]: This could be done within the `if` scope.

File: graphdriver/devmapper/deviceset.go [hunk 6]
```diff
@@ -757,6 +846,76 @@ func (devices *DeviceSet) setInitialized(hash string) error {
...
+		ids[i] = k
+		i++
+	}
+	return ids
+}
+
+func (devices *DeviceSet) deviceStatus(devName string) (sizeInSectors, mappedSectors, highestMappedSector uint64, err error) {
+	var params string
+	fmt.Println("devname", devName)
+	_, sizeInSectors, _, params, err = getStatus(devName)
```
Comment (creack) [inline graphdriver/devmapper/deviceset.go]: This could have been merged with l866

File: graphdriver/devmapper/devmapper_wrapper.go [hunk 7]
```diff
@@ -239,6 +241,58 @@ func dmTaskAddTargetFct(task *CDmTask,
 		C.uint64_t(start), C.uint64_t(size), Cttype, Cparams))
 }
 
+func getLoopbackBackingFile(file *os.File) (uint64, uint64, error) {
```
Comment (creack) [inline graphdriver/devmapper/devmapper_wrapper.go]: The idea behing devmapper_wrapper.go was to implement a 'pure' wrapper with 0 logic inside. No error checking, nothing. 100% of the lines implemented by the wrapper are used.
Would you mind moving some of the logic (error checking, loops, etc) to devmapper.go and keep only primitive in this file?

File: graphdriver/devmapper/docker-device-tool/device_tool.go [hunk 8]
```diff
@@ -1,59 +1,168 @@
 package main
```
Comment (creack) [inline graphdriver/devmapper/docker-device-tool/device_tool.go]: This file should be in /contrib or /hack. It is a tool, it should not be part of the core.

Comment (tianon) [inline graphdriver/devmapper/docker-device-tool/device_tool.go]: These seem like somewhat useful things for even regular users to be able to do - perhaps we need a way to run driver-specific commands, like `docker devicemapper:resize` ?

Comment (shykes) [inline graphdriver/devmapper/docker-device-tool/device_tool.go]: Driver-specific commands are coming :) But they will be exposed through the standard docker binary.

## 

@solomonstre
@docker

File: graphdriver/devmapper/deviceset.go (hunk 5 above)
Comment (alexlarsson) [inline graphdriver/devmapper/deviceset.go]: Not sure what you mean? the if below this is looking at err which is assigned on this line.

Comment (alexlarsson) [inline graphdriver/devmapper/deviceset.go]: Oh, you mean if foo, bar, err := f(); err != nil {}

**Comment by shykes** [comment]
LGTM

**Comment by creack** [comment]
LGTM

**Pull request merged**